src/h5web/providers/mock/mock-api.ts

```diff
@@ -39,28 +43,27 @@ export class MockApi extends ProviderApi {
       await this.cancellableDelay(params);
     }
 
-    const { value } = dataset;
+    const { value: rawValue } = dataset;
+    const value = hasArrayShape(dataset)
+      ? (rawValue as unknown[]).flat(dataset.shape.length - 1)
+      : rawValue;
     if (!selection) {
       return value;
     }
 
     assertArrayShape(dataset);
     assertPrintableType(dataset);
 
-    const dataArray = ndarray(
-      (dataset.value as (number | string | boolean)[]).flat(
-        dataset.shape.length - 1
-      ),
-      dataset.shape
-    );
-
-    const dataView = dataArray.pick(
-      ...selection
+    const { shape, type } = dataset;
+    const dataArray = ndarray(value as Primitive<typeof type>[], shape);
+    const mappedArray = applyMapping(
+      dataArray,
+      selection
         .split(',')
-        .map((s) => (s === ':' ? null : Number.parseInt(s, 10)))
+        .map((s) => (s === ':' ? 'x' : Number.parseInt(s, 10)))
     );
 
-    return unpack(dataView);
+    return mappedArray.data;
```

Comment:
I cannot use `dataView.data` as it holds the data of the full `dataArray`. I therefore make use for `applyMapping` which allow me to remove `unpack` from the project dependencies.

Reply:
Clever. Seems a bit hacky to replace `:` with `x`, though. Maybe `applyMapping` could accept the type `(':' | number)[]`? Don't think you'd have to change the implementation.